### PR TITLE
🚀 release: v2.2.0

### PR DESCRIPTION
**Pull request by warengonzaga**
## Summary by CodeRabbit

* **New Features**
  * Added customizable log formatting, allowing users to independently include or exclude ISO and local time timestamps in log outputs while always displaying log levels.
  * Introduced a new configuration option for log formatting, now accessible via the public API.

* **Documentation**
  * Updated README with comprehensive instructions and examples for customizing log element visibility and formatting options.

* **Tests**
  * Added extensive tests to verify log formatting customization, integration, and edge case handling.

* **Chores**
  * Enhanced package metadata, keywords, funding information, and improved cross-platform support for coverage uploads.

### CODE REVIEW

File: src/logger/core.ts [hunk 1]
```diff
@@ -256,7 +278,8 @@ export class Logger {
   debug(message: string, data?: LogData): void {
     if (this.shouldLog(LogLevel.DEBUG)) {
       const processedData = DataRedactor.redactData(data);
-      const formatted = LogFormatter.format(LogLevel.DEBUG, message, processedData);
+      const cachedConfig = this.getCachedConfig();
```
[the remark this answers is not in the document]
Comment (warengonzaga) [reply src/logger/core.ts:281]: What's your suggestion?
